src/sniffer/InterfaceHandler.c

```diff
@@ -0,0 +1,114 @@
+
+#include "InterfaceHandler.h"
+#include "ErrorHandler.h"
+#include "Sniffer.h"
+
+pcap_if_t *InterfaceHandler_availableInterfaces;
+pcap_if_t *InterfaceHandler_currentInterface;
+pcap_t *InterfaceHandler_captureHandler;
+
+void InterfaceHandler_Init() {
+    // Find list of interfaces available in the system
+    if(pcap_findalldevs(&InterfaceHandler_availableInterfaces, ErrorHandler_GetPcapErrorBuffer()) == PCAP_ERROR) {
+        ErrorHandler_DisplayPcapErrorAndExit("Error scanning for devices", true);
+    }
+}
+
+InterfaceHandler_status_t InterfaceHandler_SelectInterface(int interfaceNumber) {
+    int index;
+    pcap_if_t *currentIf;
+
+    if (InterfaceHandler_availableInterfaces == NULL) {
+        ErrorHandler_DisplayErrorAndExit("[InterfaceHandler] Interfaces not available!");
+    } else {
+        for (currentIf = InterfaceHandler_availableInterfaces, index = 0; currentIf != NULL; currentIf = currentIf->next, index++) {
+            if (interfaceNumber == index) {
+                InterfaceHandler_currentInterface = currentIf;
+                return InterfaceHandler_status_OK;
+            }
+        }
+    }
+
+    return InterfaceHandler_status_ERROR;
+}
+
+InterfaceHandler_status_t InterfaceHandler_OpenCapture() {
+    if (InterfaceHandler_currentInterface == NULL) {
+        ErrorHandler_DisplayWarning("[InterfaceHandler] Current interface not available!");
+        return InterfaceHandler_status_ERROR;
+    } else {
+        // Prepare capture
+        if ((InterfaceHandler_captureHandler = pcap_open_live(InterfaceHandler_currentInterface->name,	// Name of the device.
+                                BUFSIZ,			                                                        // Portion of the packet to capture. 
+                                                                                                        // Value 65536 grants that the whole packet is captured on all the MACs.
+                                1,				                                                        // Promiscuous mode.
+                                1000,			                                                        // Read timeout.
+                                ErrorHandler_GetPcapErrorBuffer()                                       // Error buffer.
+                                )) == NULL)
+        {
+            ErrorHandler_DisplayPcapErrorAndExit("Unable to open the adapter. It is not supported by Npcap", true);
+        }
+
+        //Link-layer header type values - https://www.tcpdump.org/linktypes.html
+        if (pcap_datalink(InterfaceHandler_captureHandler) != DLT_EN10MB) {
+            ErrorHandler_DisplayPcapErrorAndExit("Device does not support Ethernet headers!", false);
+        }
+
+        return InterfaceHandler_status_OK;
+    }
+}
+
+void InterfaceHandler_CapturePackets() {
+    if (InterfaceHandler_captureHandler != NULL) {
+        pcap_loop(InterfaceHandler_captureHandler, 0, Sniffer_ParsePacket, NULL);
+    }
```

Comment:
Since `InterfaceHandler_captureHandler` should never be NULL (if `pcap_open_live` fails, we exit the program), what about adding some debugging print here? This way we can know (as programmers) that we are calling this function with the variable set to NULL and the user should never see such print.

Reply:
```suggestion
    } else {
        ErrorHandler_DisplayWarning("[InterfaceHandler] Current interface not available!");
    }
```
Copied from `InterfaceHandler_StopCapturing`, feel free to use any other method